src/pages/index.tsx

```diff
@@ -1,18 +1,51 @@
-import Products from '../client/components/Products'
+import Product from '../client/components/Products'
+import { useState , useEffect} from 'react'
 import { WalletMultiButton } from '@solana/wallet-adapter-react-ui'
 import { useWallet } from '@solana/wallet-adapter-react'
+import styles from '../client/styles/HomePage.module.css'
 
 export default function HomePage() {
   const { publicKey } = useWallet()
+  const [products, setProducts] = useState<any[]>([]);
+
+  // we need to refetch the products whenever the userchanges( publickey )
```

Comment:
The comment is great, but let's make it more generic something like `Refetch products on wallet change` would work fine